### DISCUSSION TITLE
3.6.0-beta1

### DISCUSSION THREAD

**Discussion by HFO4** [cloudreve/cloudreve#1533] [General]
### 🎇Highlights

* i18n is now officially supported, with 2 built-in languages: Chinese Simplified and English (#1036, #1417, #734, #374, #671). Welcome to contribute more languages!
* **[Pro limited]** Limit available nodes by user group, users can select their preferred node before creating aria2 tasks. (#1095, #1200, #1307)

### 🎁 New Features

* Support "Seeding" status for BT download tasks, start transfer while it's still seeding. (@XYenon #1422)
* Create perm source link with shorter URL.
* Download the whole folder recursively from browser.  (@topjohncian https://github.com/cloudreve/frontend/pull/123)
* Add option to specify IP header while listening from UNIX socket, avoiding empty IP address. (@WeidiDeng #1314)
* Set TPS limit for OneDrive policy, reducing chance to hit throttle threshold.
* Gracefully shutdown server after close DB connections. (@WeidiDeng #1416)
* Show download node name in download task list page.
* Edit and remove 2FA secret for users in dashboard.
* Retry all failed upload task by one click.

### 🛠 Enhancement

* A bran new look of UI by using 12px radius as default round corner style.
* Generate error message for parameter error to give insights on which exact parameter is incorrect.
* Remove option to set ICP info in the footer, since it does not work well.
* Use file extension to search for mime type in WebDAV, resolve errors in some WebDAV client. (@WeidiDeng #1313)
* Calibrate user storage in database script, no matter if it's matching.
* Better way to remove frontend map files in build toolchain. (@AH-dark #1380)
* Release static files into memory while startup. (@WeidiDeng #1471)
* Support Retry-After throttling control from Graph API. (#280)
* Not fail immediately after failed upload in transfer tasks.
* Increase default cookie TTL to 60 days.
* Support `WAV` file as playable audio file. (@xb2016 #1246)
* Redirect to homepage while access login page with login session. (@xb2016 #1246, #1160)
* Hide sign up button if this function is disabled.  (@xb2016 #1246, #1133)
* UI for pagination, auto hiding in desktop.
* Captcha display style. (@Aruelius https://github.com/cloudreve/frontend/pull/104, https://github.com/cloudreve/frontend/pull/105)
* Use virtual list to improve performance in remote download task list. (@xb2016 https://github.com/cloudreve/frontend/pull/119)
* Hide entrance of remote download task list if this feature is disabled. (@xb2016 https://github.com/cloudreve/frontend/pull/119)
* Unified UI for torrent and URL download
* Improve path selecting experience, you can now select parent folder without extra click. (@WeidiDeng https://github.com/cloudreve/frontend/pull/138)
* Subtitle option list will be ordered by user's selected methos in file list. (@xb2016 https://github.com/cloudreve/frontend/pull/118)
* Logic to auto load subtitle with the same name with the video. (@xb2016 https://github.com/cloudreve/frontend/pull/118)

### 🐛 Bug Fixes

* Hardware authenticator is not working using FaceID on iOS devices. (#1097, #1478)
* Delete socket file before booting. (@vvisionnn #1279 #1262)
* Unexpected querying all files when deleting an empty folder. (@topjohncian #1469)
* Missing blank space in ASCII art logo.
* Admin summary cannot be cached in Redis. (@topjohncian #1329)
* Incorrect progress count in task list. (@xb2016 #1379)
* Metadata mismatch if file name contains `%` while uploading to OneDrive/SharePoint. (#1301, #1465)
* File remove operation hanging in S3 policies. (@echoface #1491, #1131)
* Cannot delete mass files (>=1000) in SQLite (#622)
* Deep copy shared header object in request options.
* Mitigate CVE-2022-32167.
* Add default cache max age for avatar response.
* [Pro] admin can purchase another user group.
* Cannot toggle dark mode in setting page. (@xb2016 #1246)
* Cannot open parent folder from remote download task list. (@xb2016 #1246)
* Crash while previewing code files. (@xb2016 #1246)
* Cards on `MyShareComponent` will not overlap and text will not be obscured when using medium screen device. (@AH-dark https://github.com/cloudreve/frontend/pull/112)
* File tiling list scroll bar. (@anzhiyu-c https://github.com/cloudreve/frontend/pull/120)
* Support unknown time zone fallback to default. (@golangaccount https://github.com/cloudreve/frontend/pull/135)
* Cannot select move destination after drag&drop. (@WeidiDeng https://github.com/cloudreve/frontend/pull/138)
* Scroll bar will appear in full screen video with large height. (@xb2016 https://github.com/cloudreve/frontend/pull/118)

<hr /><em>This discussion was created from the release <a href='https://github.com/cloudreve/Cloudreve/releases/tag/3.6.0-beta1'>3.6.0-beta1</a>.</em>

**Comment by AH-dark** [comment]
My commits #1381, #1475 are still pending review, when will it be done? 😭

**Reply by HFO4** [reply]
> My commits #1381, #1475 are still pending review, when will it be done? 😭

For #1381, it will be targeted at 3.6.1.
For #1475, I will do more investigation to see if there's better solution.

**Reply by AH-dark** [reply]
thanks.

For #1475, I think it is true that there are too many ways to configure the application, but the usual solution for containerization must be env variables. I thought so.

Of course, I'm looking forward to a better solution.
